### PR TITLE
TASK-40199: add rest end point to check if feature active for user

### PR DESCRIPTION
**Pull request by hassenelkamel-exo**

### CODE REVIEW

File: commons-component-common/src/main/java/org/exoplatform/rest/ExoFeatureRest.java [hunk 1]
```diff
@@ -0,0 +1,64 @@
+package org.exoplatform.rest;
```
Comment (boubaker) [inline commons-component-common/src/main/java/org/exoplatform/rest/ExoFeatureRest.java]: Move the file into org.exoplatform.settings.rest.ExoFeatureRest

File: commons-component-common/src/main/java/org/exoplatform/settings/impl/ExoFeatureServiceImpl.java [hunk 2]
```diff
@@ -93,14 +98,95 @@ public void changeFeatureActivation(@ManagedDescription("Feature name") @Managed
   public void addFeaturePlugin(FeaturePlugin featurePlugin) {
     plugins.put(featurePlugin.getName(), featurePlugin);
   }
-
+
```
Comment (boubaker) [inline commons-component-common/src/main/java/org/exoplatform/settings/impl/ExoFeatureServiceImpl.java]: useless space

File: commons-component-common/src/main/java/org/exoplatform/rest/ExoFeatureRest.java [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+import javax.ws.rs.PathParam;
+import javax.ws.rs.Produces;
+import javax.ws.rs.core.MediaType;
+import javax.ws.rs.core.Response;
+
+@Path("/v1/features")
+@Api(value = "/v1/features", description = "Manages exo feature associated to users")
+public class ExoFeatureRest implements ResourceContainer {
+
+  private static final String FEATURE_APP_URI = "feature";
```
Comment (boubaker) [inline commons-component-common/src/main/java/org/exoplatform/rest/ExoFeatureRest.java]: useless, to delete + format code after

File: commons-component-common/src/main/java/org/exoplatform/settings/impl/ExoFeatureServiceImpl.java [hunk 4]
```diff
@@ -93,14 +98,95 @@ public void changeFeatureActivation(@ManagedDescription("Feature name") @Managed
...
+      }
+      if (groupsOfUser == null || groupsOfUser.isEmpty()) {
+        return false;
+      }
+      for (Group group : groupsOfUser) {
+        if (permissionExpression.equals(group.getId())) {
+          return true;
+        }
+      }
+      return false;
```
Comment (boubaker) [inline commons-component-common/src/main/java/org/exoplatform/settings/impl/ExoFeatureServiceImpl.java]:
```suggestion
        try {
          Collection<Membership> memberships = organizationService.getMembershipHandler()
                                                                  .findMembershipsByUserAndGroup(username, group);
          return memberships != null && memberships.size() > 0;
        } catch (Exception e) {
          throw new IllegalStateException("Error getting memberships of user " + username, e);
        }
```

File: commons-component-common/src/main/java/org/exoplatform/settings/impl/ExoFeatureServiceImpl.java (hunk 2 above)
Comment (hassenelkamel-exo) [reply commons-component-common/src/main/java/org/exoplatform/settings/impl/ExoFeatureServiceImpl.java]: it is a one line

Comment (boubaker) [reply commons-component-common/src/main/java/org/exoplatform/settings/impl/ExoFeatureServiceImpl.java]:
```suggestion

```

File: commons-component-common/src/main/java/org/exoplatform/settings/rest/ExoFeatureRest.java [hunk 5]
```diff
@@ -0,0 +1,62 @@
...
+import javax.annotation.security.RolesAllowed;
+import javax.ws.rs.GET;
+import javax.ws.rs.Path;
+import javax.ws.rs.PathParam;
+import javax.ws.rs.Produces;
+import javax.ws.rs.core.MediaType;
+import javax.ws.rs.core.Response;
+
+@Path("/v1/features")
+@Api(value = "/v1/features", description = "Manages exo feature associated to users")
```
Comment (boubaker) [inline commons-component-common/src/main/java/org/exoplatform/settings/rest/ExoFeatureRest.java]:
```suggestion
@Api(value = "/v1/features", description = "Manages product experimental features")
```

File: commons-extension-webapp/src/main/webapp/WEB-INF/conf/commons-extension/feature-configuration.xml [hunk 6]
```diff
@@ -0,0 +1,17 @@
...
+  xmlns="http://www.exoplatform.org/xml/ns/kernel_1_2.xsd">
+
+    <component profiles="all">
+        <key>org.exoplatform.commons.api.settings.ExoFeatureService</key>
+        <type>org.exoplatform.settings.impl.ExoFeatureServiceImpl</type>
+    </component>
+
+    <component profiles="all">
+        <type>org.exoplatform.settings.rest.ExoFeatureRest</type>
+    </component>
```
Comment (boubaker) [inline commons-extension-webapp/src/main/webapp/WEB-INF/conf/commons-extension/feature-configuration.xml]:
```suggestion
    <component>
        <key>org.exoplatform.commons.api.settings.ExoFeatureService</key>
        <type>org.exoplatform.settings.impl.ExoFeatureServiceImpl</type>
    </component>

    <component>
        <type>org.exoplatform.settings.rest.ExoFeatureRest</type>
    </component>
```

File: commons-component-common/src/main/java/org/exoplatform/settings/impl/ExoFeatureServiceImpl.java [hunk 7]
```diff
@@ -93,14 +98,87 @@ public void changeFeatureActivation(@ManagedDescription("Feature name") @Managed
...
+
+    } else if (permissionExpression.contains("/")) {
+      String group = permissionExpression;
+      org.exoplatform.services.security.Identity identity = identityRegistry.getIdentity(username);
+      if (identity != null) {
+        return identity.isMemberOf(permissionExpression);
+      }
+      try {
+        Collection<Membership> memberships = organizationService.getMembershipHandler()
+                                                                .findMembershipsByUserAndGroup(username, group);
```
Comment (boubaker) [inline commons-component-common/src/main/java/org/exoplatform/settings/impl/ExoFeatureServiceImpl.java]:
```suggestion
        Collection<Membership> memberships = organizationService.getMembershipHandler()
                                                                .findMembershipsByUserAndGroup(username, permissionExpression);
```

File: commons-component-common/src/main/java/org/exoplatform/settings/impl/ExoFeatureServiceImpl.java [hunk 8]
```diff
@@ -93,14 +98,87 @@ public void changeFeatureActivation(@ManagedDescription("Feature name") @Managed
...
+    } else {
+      String propertyValue = System.getProperty(propertyName);
+      Boolean active = propertyValue != null ? Boolean.valueOf(propertyValue) : null;
+      featuresProperties.put(propertyName, active);
+      return active;
+    }
+  }
+
+  private List<String> getFeaturePermissionPropertyValues(String featureName) {
+    String propertyName = "exo.feature.agenda." + featureName + ".permissions";
```
Comment (boubaker) [inline commons-component-common/src/main/java/org/exoplatform/settings/impl/ExoFeatureServiceImpl.java]:
```suggestion
    String propertyName = "exo.feature." + featureName + ".permissions";
```

**Review by boubaker — DISMISSED** [review]

**Review by boubaker — APPROVED** [review]

**Pull request merged**